docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+```yaml
+# Payment service worker
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: payment-processor
+  namespace: payments
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: payments
+  cutover:
+    strategy: Progressive  # Conservative rollout for financial operations
+    steps:
+      - rampPercentage: 5
+        pauseDuration: 10m
+      - rampPercentage: 25
+        pauseDuration: 15m
+  # ... rest of config
+---
+# Notification service worker (separate service, different risk profile)
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: notification-sender
+  namespace: notifications
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: notifications
+  cutover:
+    strategy: AllAtOnce  # Lower risk, faster rollouts acceptable
+  # ... rest of config
+```
+
+### Pattern 2: Environment-Specific Strategies
+
+Use different rollout strategies based on environment risk:
+
+```yaml
+# Production - Conservative rollout
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: order-processor
+  namespace: production
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: production
+  cutover:
+    strategy: Progressive
+    steps:
+      - rampPercentage: 10
+        pauseDuration: 15m
+      - rampPercentage: 50
+        pauseDuration: 30m
+    gate:
+      workflowType: "HealthCheck"  # Validate new version before proceeding
+---
+# Staging - Fast rollout for testing
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: order-processor
+  namespace: staging
+spec:
+  workerOptions:
+    connection: staging-temporal
+    temporalNamespace: staging
+  cutover:
+    strategy: AllAtOnce  # Immediate cutover for faster iteration
+```
+
+### Pattern 3: Gradual Team Migration
+
+Migrate teams/services based on their readiness and risk tolerance:
+
+**Phase 1: Low-Risk Services**
+- Background processing workers
+- Internal tooling workflows
+- Non-customer-facing operations
+
+**Phase 2: Medium-Risk Services**  
+- Internal API workflows
+- Data processing pipelines
+- Administrative workflows
+
+**Phase 3: High-Risk Services**
+- Customer-facing workflows
+- Payment processing
+- Critical business operations
+
+
+
+## Troubleshooting
+
+### Common Issues
+
+**1. Workers Not Registering with Temporal**
+
+*Symptoms:*
+```
+status:
+  targetVersion:
+    status: NotRegistered
+```
+
+*Solutions:*
+- Check worker logs for connection errors
+- Verify TemporalConnection configuration  
+- Ensure TLS secrets are properly configured
+- Verify network connectivity to Temporal server
+- Check that worker code properly handles `WORKER_BUILD_ID` environment variable
+
+**2. New Workflows Still Going to Unversioned Workers**
+
+*Symptoms:*
+- Temporal UI shows workflows executing on unversioned workers
+- Versioned workers appear idle
```

Comment:
not sure what "appear idle" would look like, but the Temporal UI shows workflows *not* executing on versioned workers (you can see this by clicking "workflows" from the version line item in the UI)